source/appModules/foobar2000.py

```diff
@@ -1,25 +1,41 @@
-import appModuleHandler
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2009-2018 NV Access Limited, Aleksey Sadovoy, James Teh, Joseph Lee, Tuukka Ojala
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.htmlimport appModuleHandler
+
+import calendar
+import time
+
 import api
+import appModuleHandler
 import ui
-import time
-import calendar
 
-def getFormat(s):
-	s=s.split(":")
-	if len(s) ==1:
+def getParsingFormat(interval):
+	"""Attempts to find a suitable parsing format string for a HH:MM:SS, MM:SS or SS -style time interval."""
+	timeParts = len(interval.split(":"))
+	if timeParts == 1:
 		return "%S"
-	elif len(s)==2:
+	elif timeParts == 2:
 		return "%M:%S"
-	else: return "%H:%M:%S"
+	elif timeParts == 3:
+		return "%H:%M:%S"
+	else:
+		return None
 
-def getOutputFormat(t):
-	if t<60:
+def getOutputFormat(seconds):
+	"""Returns a format string for the given number of seconds with the least leading zeros."""
+	if seconds < 60:
 		return "%S"
-	elif t<3600:
+	elif seconds < 3600:
 		return "%M:%S"
-	else: 
+	else:
 		return "%H:%M:%S"
 
+def parseIntervalToTimestamp(interval):
+	"""Parses a HH:MM:SS, MM:SS or SS -style interval to a timestamp."""
+	format = getParsingFormat(interval)
+	return calendar.timegm(time.strptime(interval.strip(), getParsingFormat(interval)))

```

Comment:
Why do you have 
getParsingFormat(interval)))
here, when format is a variable?